projects/components/src/form/number-with-unit-input/number-with-unit-form-input.component.spec.ts

```diff
@@ -137,6 +137,32 @@ describe('VcdNumberWithUnitFormInputComponent', () => {
             numberWithUnitInput.clickUnlimitedCheckbox();
             expect(numberWithUnitInput.isInputValueFocused).toBe(true, 'Input element should have focus');
         });
+
+        it('will choose a unit if initially unlimited when unchecked', () => {
```

Comment:
not clear isn't it better
will choose a unit when unchecked for initial value unlimited